web/src/components/Editor/Editor.tsx

```diff
@@ -46,16 +47,23 @@ const Editor = forwardRef((props: Props, ref: React.ForwardedRef<EditorRefAction
       focus: () => {
         editorRef.current?.focus();
       },
-      insertText: (rawText: string) => {
+      insertText: (rawText: string, mark = false) => {
         if (!editorRef.current) {
           return;
         }
+        const cursorPosition = editorRef.current.selectionStart;
+        const endPosition = editorRef.current.selectionEnd;
+        if (endPosition === cursorPosition && mark) return;
 
         const prevValue = editorRef.current.value;
-        const cursorPosition = editorRef.current.selectionStart;
-        editorRef.current.value = prevValue.slice(0, cursorPosition) + rawText + prevValue.slice(cursorPosition);
+
+        let value = prevValue.slice(0, cursorPosition) + rawText;
+        if (mark) {
+          value += prevValue.slice(cursorPosition, endPosition) + rawText + prevValue.slice(endPosition);
+        }
+        editorRef.current.value = value;
```

Comment:
Using the insert function below the editor after selecting the text will lose the selected text